Networking/Networking/Mapper/MetaDataMapper.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+/// Mapper: MetaData List
+///
+struct MetaDataMapper: Mapper {
+
+    /// (Attempts) to convert an instance of Data into an array of MetaData Entities.
+    ///
+    func map(response: Data) throws -> [MetaData] {
+        let decoder = JSONDecoder()
+        if hasDataEnvelope(in: response) {
+            let envelope = try decoder.decode(DataEnvelope.self, from: response)
+            return envelope.data.metadata.filter { !$0.key.hasPrefix("_") }
+        } else {
+            let envelope = try decoder.decode(MetaDataEnvelope.self, from: response)
+            return envelope.metadata.filter { !$0.key.hasPrefix("_") }
+        }
```

Comment:
I think we can do the following to prevent having the filter statement twice. 

```suggestion
         let metadata = try {
            if hasDataEnvelope(in: response) {
                let envelope = try decoder.decode(DataEnvelope.self, from: response)
                return envelope.data.metadata
            } else {
                let envelope = try decoder.decode(MetaDataEnvelope.self, from: response)
                return envelope.metadata
            }
        }()

        return metadata.filter { !$0.key.hasPrefix("_") }
```

Reply:
Done a5848d37fef0afcbfdd87b8aef75564e0bc72829 👍 thanks for the review Sharma!